### PR TITLE
Fix infinite loop when encountering unexpected test specifications

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

**Comment by Marenz** [comment]
the test causing a inf loop would look like this:
```
contract C {
    function f() public pure returns (uint16 x) {
    uint8 y;
    assembly {
        y := 0x12345678
    }
    x = y;
    }
}
// ====
// compileViaYul: true
// ----
// f() -> ?? # 
```

**Comment by Marenz** [comment]
added test

File: test/libsolidity/util/TestFileParserTests.cpp [hunk 1]
```diff
@@ -775,6 +775,14 @@ BOOST_AUTO_TEST_CASE(call_arrow_missing)
 	BOOST_REQUIRE_THROW(parse(source), langutil::Error);
 }
 
+BOOST_AUTO_TEST_CASE(inf_loop_bug)
```
Comment (erak) [inline test/libsolidity/util/TestFileParserTests.cpp]:
```suggestion
BOOST_AUTO_TEST_CASE(call_unexpected_character)
```

**Comment by Marenz** [comment]
renamed test

**Review by erak — APPROVED** [review]

**Pull request merged**